kyaml/fn/framework/framework.go

```diff
@@ -170,3 +321,120 @@ func (rl *ResourceList) Filter(api kio.Filter) error {
 	}
 	return nil
 }
+
+// SortItems sorts the ResourceList.items by apiVersion, kind, namespace and name.
+func (rl *ResourceList) SortItems() {
+	sort.Sort(sortRNodeObjects(rl.Items))
+}
+
+type sortRNodeObjects []*yaml.RNode
+
+func (o sortRNodeObjects) Len() int      { return len(o) }
+func (o sortRNodeObjects) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o sortRNodeObjects) Less(i, j int) bool {
+	idi := o[i].GetResourceIdentifier()
+	idj := o[j].GetResourceIdentifier()
+	idStrI := fmt.Sprintf("%s %s %s %s", idi.GetAPIVersion(), idi.GetKind(), idi.GetNamespace(), idi.GetName())
+	idStrJ := fmt.Sprintf("%s %s %s %s", idj.GetAPIVersion(), idj.GetKind(), idj.GetNamespace(), idj.GetName())
+	return idStrI < idStrJ
+}
+
+// Run evaluates the function. input must be a resourceList in yaml format. An
+// updated resourceList will be returned.
+func Run(p ResourceListProcessor, input []byte) ([]byte, error) {
+	rl, err := ParseResourceList(input)
+	if err != nil {
+		return nil, err
+	}
+	err = p.Process(rl)
+	if err != nil {
+		// If the error is not a Results type, we wrap the error as a Result.
+		if results, ok := err.(Results); ok {
+			rl.Results = append(rl.Results, results...)
+		} else if result, ok := err.(Result); ok {
+			rl.Results = append(rl.Results, &result)
+		} else if result, ok := err.(*Result); ok {
+			rl.Results = append(rl.Results, result)
+		} else {
+			rl.Results = append(rl.Results, ErrorResult(err))
+		}
+	}
+	yml, er := rl.ToYAML()
+	if er != nil {
+		return []byte(yml), er
+	}
+	if len(rl.Results) > 0 {
+		return []byte(yml), rl.Results
+	}
+	return []byte(yml), nil
+}
+
+type ErrMissingFnConfig struct{}
+
+func (ErrMissingFnConfig) Error() string {
+	return "unable to find the functionConfig in the resourceList"
+}
+
+// Chain chains a list of ResourceListProcessor as a single ResourceListProcessor.
+func Chain(processors ...ResourceListProcessor) ResourceListProcessor {
+	return ResourceListProcessorFunc(func(rl *ResourceList) error {
+		for _, processor := range processors {
+			if err := processor.Process(rl); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+// ChainFunctions chains a list of ResourceListProcessorFunc as a single ResourceListProcessorFunc.
+func ChainFunctions(fns ...ResourceListProcessorFunc) ResourceListProcessorFunc {
```

Comment:
Similar question--not used in tests, and I'm also unsure why we'd need this in addition to `Chain` given that `ResourceListProcessorFunc` _are_ `ResourceListProcessor`.